Sources/MapboxNavigation/NavigationMapView+VanishingRouteLine.swift

```diff
@@ -202,34 +202,54 @@ extension NavigationMapView {
         if traveledDifference == 0.0 {
             return
         }
-        let startDate = Date()
-        vanishingRouteLineUpdateTimer = Timer.scheduledTimer(withTimeInterval: 0.05, repeats: true, block: { timer in
-            let timePassedInMilliseconds = Date().timeIntervalSince(startDate) * 1000
-            if timePassedInMilliseconds >= 980 {
-                timer.invalidate()
-                return
-            }
-            
-            let newFractionTraveled = self.preFractionTraveled + traveledDifference * timePassedInMilliseconds.truncatingRemainder(dividingBy: 1000) / 1000
-            
+        switch puckType {
+        case .default:
+            let startDate = Date()
+            vanishingRouteLineUpdateTimer = Timer.scheduledTimer(withTimeInterval: 0.05, repeats: true, block: { timer in
+                let timePassedInMilliseconds = Date().timeIntervalSince(startDate) * 1000
+                if timePassedInMilliseconds >= 980 {
```

Comment:
I overlooked it before, but this code should probably have more comments explaining what’s significant about 980&nbsp;milliseconds, among other things.

Reply:
Because in case if the time was already in the last interval, it would invalidate the timer and wait for the next routeProgress update. I'll add comment explaining this one. Thanks for the suggestion.

Reply:
BTW this functionality was added when we were not updating location puck frame by frame. Since new camera functionality requires frame by frame updates we can probably re-evaluate current implementation and simplify it by using MapView callbacks instead of such timers.